features/LimitReuse.php

```diff
@@ -44,147 +45,107 @@ public function __construct(BetterPasswordPlugin $plugin) {
-				//$user = Repo::user()->getByUsername($username);
-				$user = $this->_addPassword($user, $currentPassword);
-				$this->_handledPasswordUpdate = true;
-				//$userDao->updateObject($user);
-				Repo::user()->edit($user);
-			});
-		}; 
-		
-		//Hook::add('changepasswordform::execute', [$this, 'rememberPasswords']);
-		Hook::add('changepasswordform::execute', [$this, 'rememberPasswords']);
-		//Hook::add('LoadComponentHandler', $handler);
-		//Hook::add('LoadHandler', $handler);*/
-	}
-
 	public function addToSchema($hook, $args) {
 		$user = $args[0];
 
-		$user->properties->previousPasswords = (object) [
+		$user->properties->{$this->_plugin->getSettingsName()."::lastPasswords"} = (object) [
 			'type' => 'string',
 			'apiSummary' => false,
-			//'multilingual' => false,
 			'validation' => ['nullable']
 		];
 
-		$user->properties->previousPasswordsUpdate = (object) [
+		$user->properties->{$this->_plugin->getSettingsName()."::lastPasswordsUpdate"} = (object) [
 			'type' => 'string',
 			'apiSummary' => false,
-			//'multilingual' => false,
 			'validation' => ['nullable', 'date:Y-m-d H:i:s']
 		];
 
 		return false;
 	}
 
+	/**
+	 * Checks the current form and remembers the user's passwords
+	 * @param string $hook The hook name
+	 * @param array $args Arguments of the hook
+	 * @return bool false if process completes
+	 */
 	public function rememberPasswords($hook, $args) {
 		if ($hook == 'changepasswordform::execute' && get_class($args[0]) == 'PKP\user\form\ChangePasswordForm') {
 			$user = $args[0]->getUser();
-			$password = $args[0]->getData('oldPassword');
-			$user = $this->_addPassword($user, $password);
-			//Repo::user()->edit($user);
-			Repo::user()->edit($user);
-			//put assignment of objects here
 		}
+		else if ($hook == 'loginchangepasswordform::execute' && get_class($args[0]) == 'PKP\user\form\LoginChangePasswordForm'){
+			$user = Repo::user()->getByUsername($args[0]->getData('username'), false);
+		}
+
+		$password = $user->getData('password');
+		$this->_addPassword($user, $password);
 		return false;
 	}
 
+	/**
+	 * Checks the current form and checks if the user's inputted password has been used previously
+	 * @param string $hook The hook name
+	 * @param array $args Arguments of the hook
+	 * @return bool true if process completes
+	 */
 	public function passwordChangeValidation($hook, $args) {
 		$form = $args[0];
-		$newCheck = new FormValidatorCustom($form, 'password', 'required', 'plugins.generic.betterPassword.validation.betterPasswordPasswordReused', $this->passwordCompare($password, $form));
+		$newCheck = new FormValidatorCustom($form, 'password', 'required', 'plugins.generic.betterPassword.validation.betterPasswordPasswordReused', function ($password) use ($form) {
+			 return $this->passwordCompare($password, $form);
+		}); //$this->passwordCompare());
 		$form->addCheck($newCheck);
 		return false;
 	}
 
+	/**
+	 * Updates the time of the users last password change
```

Comment:
This appears to be comparing the password against prior passwords, not update the time.